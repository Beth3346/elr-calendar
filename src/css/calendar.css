.elr-calendar-view-nav,
.elr-calendar-nav {
  margin-bottom: 10px;
  display: flex;
}

.elr-calendar-view-nav button:first-child,
.elr-calendar-nav button:first-child {
  border-radius: 3px 0 0 3px;
}

.elr-calendar-view-nav button:last-child,
.elr-calendar-nav button:last-child {
  border-radius: 0 3px 3px 0;
}

.elr-calendar-view-nav button,
.elr-calendar-nav button {
  display: block;
  min-width: 150px;
  cursor: pointer;
  background-color: #333;
  color: #fff;
  text-align: center;
  padding: 10px 15px;
  font-size: 14px;
  font-weight: 700;
  transition: background-color 0.3s;
}

.elr-calendar-view-nav button:hover,
.elr-calendar-nav button:hover {
  background-color: #444;
}

.elr-calendar-view-nav button.active {
  background-color: #444;
}

.elr-month {
  border: 1px solid #eee;
}

.elr-week.even-week {
  background-color: mistyrose;
}

.elr-calendar-heading {
  display: flex;
  justify-content: space-around;
}

.elr-week {
  display: flex;
  justify-content: space-around;
  height: 100px;
  border-bottom: 1px solid #eee;
}

.elr-week:last-of-type {
  border-bottom: 0;
}

.elr-week > div {
  flex-shrink: 1;
  flex-grow: 1;
  padding: 5px;
  flex: 1;
  /*background-color: red;*/
}

.elr-cal-muted {
  background-color: #eee;
}

.elr-cal-weekend {
  background-color: lightpink;
}

.elr-cal-today {
  background-color: paleturquoise;
}

.elr-event-list {
  list-style: none;
}

.elr-event-list li a {
  font-size: 13px;
}
