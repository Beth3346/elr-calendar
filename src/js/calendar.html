<div class="elr-calendar">
  <div class="elr-calendar-view-nav elr-button-group">
    <button class="elr-button elr-button-primary elr-calendar-view" data-view="month">Month</button>
    <button class="elr-button elr-button-primary elr-calendar-view" data-view="week">Week</button>
    <button class="elr-button elr-button-primary elr-calendar-view" data-view="date">Day</button>
  </div>
  <div class="elr-calendar-nav elr-button-group">
    <button class="elr-button elr-button-primary elr-calendar-year-prev" data-dir="prev">Previous Year</button>
    <button class="elr-button elr-button-primary elr-calendar-month-prev" data-dir="prev">Previous Month</button>
    <button class="elr-button elr-button-primary elr-calendar-week-prev" data-dir="prev">Previous Week</button>
    <button class="elr-button elr-button-primary elr-calendar-date-prev" data-dir="prev">Yesterday</button>
    <button class="elr-button elr-button-primary elr-calendar-current">Today</button>
    <button class="elr-button elr-button-primary elr-calendar-date-next" data-dir="next">Tomorrow</button>
    <button class="elr-button elr-button-primary elr-calendar-week-next" data-dir="next">Next Week</button>
    <button class="elr-button elr-button-primary elr-calendar-month-next" data-dir="next">Next Month</button>
    <button class="elr-button elr-button-primary elr-calendar-year-next" data-dir="next">Next Year</button>
  </div>
</div>
