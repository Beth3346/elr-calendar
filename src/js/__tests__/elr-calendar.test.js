import $ from 'jquery'
import calendar from '../elr-calendar'
import fs from 'fs'
import path from 'path'
import time from 'elr-time-utilities'

jest.dontMock('fs').dontMock('jquery')

const calendarHtml = path.join(__dirname, '..', 'calendar.html')
const html = fs.readFileSync(calendarHtml).toString()

beforeEach(() => {
  document.body.innerHTML = html
})

describe('init', () => {
  it('should render a calendar', () => {
    calendar.init()
    expect($('.elr-calendar-month-view').length).toBe(1)
  })
  it('should take parameters', () => {
    calendar.init({
      view: 'week'
    })
    expect($('.elr-calendar-week-view').length).toBe(1)
  })
  it('should render a provided day', () => {
    calendar.init({
      view: 'month',
      currentDate: {
        month: 3,
        date: 1,
        year: 2014
      }
    })
    expect($('.elr-calendar-month-view').data('month')).toBe(3)
    expect($('.elr-calendar-month-view').data('year')).toBe(2014)
  })
  it('should show holidays', () => {
    calendar.init({
      view: 'month',
      currentDate: {
        month: 2,
        date: 1,
        year: 2018
      }
    })
    expect($('.elr-cal-holiday').length).toBe(1)
    expect($('.elr-cal-holiday').html()).toMatchSnapshot()
  })
  it('should show floating holidays', () => {
    calendar.init({
      view: 'month',
      currentDate: {
        month: 0,
        date: 1,
        year: 2018
      }
    })
    expect($('#date0152018').find('.elr-cal-holiday').length).toBe(1)
    expect(
      $('#date0152018')
        .find('.elr-cal-holiday')
        .html()
    ).toMatchSnapshot()
  })
  it('should advance year', () => {
    calendar.init({
      view: 'month',
      currentDate: {
        month: 3,
        date: 1,
        year: 2014
      }
    })

    $('.elr-calendar-year-next').click()

    expect($('.elr-calendar-month-view').data('year')).toBe(2015)

    $('.elr-calendar-year-next').click()

    expect($('.elr-calendar-month-view').data('year')).toBe(2016)

    $('.elr-calendar-year-prev').click()

    expect($('.elr-calendar-month-view').data('year')).toBe(2015)

    $('.elr-calendar-year-prev').click()

    expect($('.elr-calendar-month-view').data('year')).toBe(2014)

    $('.elr-calendar-year-prev').click()

    expect($('.elr-calendar-month-view').data('year')).toBe(2013)
  })
  it('should advance month', () => {
    calendar.init({
      view: 'month',
      currentDate: {
        month: 1,
        date: 1,
        year: 2014
      }
    })

    $('.elr-calendar-month-next').click()

    expect($('.elr-calendar-month-view').data('month')).toBe(2)

    $('.elr-calendar-month-next').click()

    expect($('.elr-calendar-month-view').data('month')).toBe(3)

    $('.elr-calendar-month-prev').click()

    expect($('.elr-calendar-month-view').data('month')).toBe(2)
  })
  it('should show current date', () => {
    calendar.init({
      view: 'month',
      currentDate: {
        month: 1,
        date: 1,
        year: 2014
      }
    })

    $('.elr-calendar-month-next').click()

    expect($('.elr-calendar-month-view').data('month')).toBe(2)

    $('.elr-calendar-year-next').click()

    expect($('.elr-calendar-month-view').data('year')).toBe(2015)

    $('.elr-calendar-current').click()

    expect($('.elr-calendar-month-view').data('month')).toBe(1)
    expect($('.elr-calendar-month-view').data('year')).toBe(2014)
  })
  it('should go to december on prev month click if its january', () => {
    calendar.init({
      view: 'month',
      currentDate: {
        month: 0,
        date: 1,
        year: 2014
      }
    })

    $('.elr-calendar-month-prev').click()

    expect($('.elr-calendar-month-view').data('month')).toBe(11)
    expect($('.elr-calendar-month-view').data('year')).toBe(2013)
  })
  it('should go to january on next month click if its december', () => {
    calendar.init({
      view: 'month',
      currentDate: {
        month: 11,
        date: 1,
        year: 2014
      }
    })

    $('.elr-calendar-month-next').click()

    expect($('.elr-calendar-month-view').data('month')).toBe(0)
    expect($('.elr-calendar-month-view').data('year')).toBe(2015)
  })
  it('should display events that occur on a monthly basis', () => {
    calendar.init({
      view: 'month',
      currentDate: {
        month: 1,
        date: 1,
        year: 2018
      },
      newEvents: [
        {
          name: 'First Tuesday',
          recurrance: 'monthly',
          day: ['Tuesday'],
          dayNum: 1
        }
      ]
    })
    expect($('#date162018').find('ul.elr-event-list').length).toBe(1)
    expect(
      $('#date162018')
        .find('ul.elr-event-list')
        .html()
    ).toMatchSnapshot()
  })
  it('should add a class to weekends', () => {
    calendar.init({
      view: 'month',
      currentDate: {
        month: 1,
        date: 1,
        year: 2018
      }
    })
    expect($('#date142018').hasClass('elr-cal-weekend')).toBe(true)
  })
  xit('should add class to current date', () => {
    const today = time().today()
    const id = `#date${today.month}${today.date}${today.year}`
    calendar.init({
      view: 'month',
      currentDate: today
    })

    expect($(id).hasClass('elr-cal-today')).toBe(true)
  })
  it('should add week numbers', () => {
    calendar.init({
      view: 'month',
      currentDate: {
        month: 1,
        date: 1,
        year: 2018
      }
    })
    expect(
      $('.elr-week')
        .eq(1)
        .data('week')
    ).toBe(6)
  })
  it('should add odd/even week classes', () => {
    calendar.init({
      view: 'month',
      currentDate: {
        month: 1,
        date: 1,
        year: 2018
      }
    })
    expect(
      $('.elr-week')
        .eq(1)
        .hasClass('even-week')
    ).toBeTruthy()
    expect(
      $('.elr-week')
        .eq(2)
        .hasClass('odd-week')
    ).toBeTruthy()
  })
  it('should show days of the week', () => {
    calendar.init({
      view: 'month',
      currentDate: {
        month: 1,
        date: 1,
        year: 2018
      }
    })
    expect($('.elr-calendar .elr-calendar-heading').length).toBe(1)
    expect($('.elr-calendar .elr-calendar-heading').html()).toMatchSnapshot()
  })
  it('should create a date element', () => {
    calendar.init({
      view: 'month',
      currentDate: {
        month: 1,
        date: 1,
        year: 2018
      }
    })
    expect(
      $('.elr-calendar .elr-week:nth-child(2) > div:first-child').html()
    ).toMatchSnapshot()
  })
  it('should create the first week of a month', () => {
    calendar.init({
      view: 'month',
      currentDate: {
        month: 1,
        date: 1,
        year: 2018
      }
    })
    expect($('.elr-calendar .elr-week:nth-child(1)').html()).toMatchSnapshot()
  })
  it('should create the last week of a month', () => {
    calendar.init({
      view: 'month',
      currentDate: {
        month: 1,
        date: 1,
        year: 2018
      }
    })
    expect($('.elr-calendar .elr-week:last-of-type').html()).toMatchSnapshot()
  })
})
