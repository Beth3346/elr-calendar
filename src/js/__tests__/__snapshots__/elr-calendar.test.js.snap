// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init should create a date element 1`] = `"4"`;

exports[`init should create the first week of a month 1`] = `"<div id=\\"date0282018\\" class=\\"elr-cal-muted\\" data-month=\\"0\\" data-date=\\"28\\" data-year=\\"2018\\" data-day=\\"0\\">28</div><div id=\\"date0292018\\" class=\\"elr-cal-muted\\" data-month=\\"0\\" data-date=\\"29\\" data-year=\\"2018\\" data-day=\\"1\\">29</div><div id=\\"date0302018\\" class=\\"elr-cal-muted\\" data-month=\\"0\\" data-date=\\"30\\" data-year=\\"2018\\" data-day=\\"2\\">30</div><div id=\\"date0312018\\" class=\\"elr-cal-muted\\" data-month=\\"0\\" data-date=\\"31\\" data-year=\\"2018\\" data-day=\\"3\\">31</div><div id=\\"date112018\\" class=\\"elr-date\\" data-month=\\"1\\" data-date=\\"1\\" data-year=\\"2018\\" data-day=\\"4\\">1</div><div id=\\"date122018\\" class=\\"elr-date\\" data-month=\\"1\\" data-date=\\"2\\" data-year=\\"2018\\" data-day=\\"5\\">2<ul class=\\"elr-event-list\\"><li><a href=\\"#\\" class=\\"elr-cal-holiday\\"><span class=\\"elr-event elr-all-day-event\\">Groundhog Day</span></a></li></ul></div><div id=\\"date132018\\" class=\\"elr-date elr-cal-weekend\\" data-month=\\"1\\" data-date=\\"3\\" data-year=\\"2018\\" data-day=\\"6\\">3</div>"`;

exports[`init should create the last week of a month 1`] = `"<div id=\\"date1252018\\" class=\\"elr-date elr-cal-weekend\\" data-month=\\"1\\" data-date=\\"25\\" data-year=\\"2018\\" data-day=\\"0\\">25</div><div id=\\"date1262018\\" class=\\"elr-date\\" data-month=\\"1\\" data-date=\\"26\\" data-year=\\"2018\\" data-day=\\"1\\">26</div><div id=\\"date1272018\\" class=\\"elr-date\\" data-month=\\"1\\" data-date=\\"27\\" data-year=\\"2018\\" data-day=\\"2\\">27</div><div id=\\"date1282018\\" class=\\"elr-date\\" data-month=\\"1\\" data-date=\\"28\\" data-year=\\"2018\\" data-day=\\"3\\">28</div><div id=\\"date212018\\" class=\\"elr-cal-muted\\" data-month=\\"2\\" data-date=\\"1\\" data-year=\\"2018\\" data-day=\\"4\\">1</div><div id=\\"date222018\\" class=\\"elr-cal-muted\\" data-month=\\"2\\" data-date=\\"2\\" data-year=\\"2018\\" data-day=\\"5\\">2</div><div id=\\"date232018\\" class=\\"elr-cal-muted\\" data-month=\\"2\\" data-date=\\"3\\" data-year=\\"2018\\" data-day=\\"6\\">3</div>"`;

exports[`init should display events that occur on a monthly basis 1`] = `"<li><a href=\\"#\\"><span class=\\"elr-event elr-all-day-event\\">First Tuesday</span></a></li>"`;

exports[`init should show days of the week 1`] = `"<div class=\\"elr-calendar-heading-day\\">Sunday</div><div class=\\"elr-calendar-heading-day\\">Monday</div><div class=\\"elr-calendar-heading-day\\">Tuesday</div><div class=\\"elr-calendar-heading-day\\">Wednesday</div><div class=\\"elr-calendar-heading-day\\">Thursday</div><div class=\\"elr-calendar-heading-day\\">Friday</div><div class=\\"elr-calendar-heading-day\\">Saturday</div>"`;

exports[`init should show floating holidays 1`] = `"<span class=\\"elr-event elr-all-day-event\\">Martin Luther King's Birthday</span>"`;

exports[`init should show holidays 1`] = `"<span class=\\"elr-event elr-all-day-event\\">St. Patrick's Day</span>"`;
